### PR TITLE
Share into prefix

### PR DESCRIPTION
**Pull request by harshad1**
This PR addresses #1368.

1. Added a field in settings for the prefix
2. Prefix is added to any text shared into app
3. Prefix + shared text is shown in share into dialog so user can edit it as needed

### CODE REVIEW

**Comment by gsantner** [comment]
Suggestion:

* Naming - ShareInto Format
* Usage - With %s. So `\n---\n%s\n---` inserts in the middle

what do you think?

File: app/src/main/java/net/gsantner/markor/activity/DocumentShareIntoFragment.java [hunk 1]
```diff
@@ -99,17 +103,28 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
...
         _hlEditor.setTextSize(TypedValue.COMPLEX_UNIT_SP, as.getFontSize());
         _hlEditor.setTypeface(Typeface.create(as.getFontFamily(), Typeface.NORMAL));
 
-        if (sharedText.trim().isEmpty()) {
+        if (sharedText.isEmpty() || sharedText.equals(prefix)) {
             _hlEditor.requestFocus();
         }
     }
 
+    private static String formatOrPrefix(final String format, final String value) {
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/markor/activity/DocumentShareIntoFragment.java:118]: simple:
```
format = format.contains("%s") ? format : (format + "%s");
result = format.replace("%s", value);
```

doesn't need real string formatting, just a token which is %s

File: app/src/main/java/net/gsantner/markor/activity/DocumentShareIntoFragment.java [hunk 2]
```diff
@@ -99,17 +103,28 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
...
+        try {
+            final String result  = String.format(format, value);
+            if (!result.equals(format) && !TextUtils.isEmpty(value)) {
+                return result;
+            }
+        } catch (IllegalFormatException ignored) { }
+        return format + " " + value;
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/markor/activity/DocumentShareIntoFragment.java:125]:
```suggestion
String fmt = format.contains("%s") ? format : (format + "%s");
return fmt.replace("%s", TextUtils.isEmpty(value) ? "" : value);
```

File: app/src/main/java/net/gsantner/markor/activity/DocumentShareIntoFragment.java [hunk 3]
```diff
@@ -99,17 +103,28 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
...
+    }
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/markor/activity/DocumentShareIntoFragment.java:126]:
```suggestion
    private static String formatOrPrefix(String format, final String value) {
        format = format.contains("%s") ? format : (format + "%s");
        return format.replace("%s", TextUtils.isEmpty(value) ? "" : value);
    }
```

Comment (harshad1) [reply app/src/main/java/net/gsantner/markor/activity/DocumentShareIntoFragment.java:126]: I thought of this, but felt that there were edge cases with escaped '\%s' etc. In the end It's only a couple lines extra

File: app/src/main/java/net/gsantner/markor/activity/DocumentShareIntoFragment.java [hunk 4]
```diff
@@ -116,13 +116,7 @@ public void onViewCreated(final @NonNull View view, final Bundle savedInstanceSt
...
 
     private static String formatOrPrefix(final String format, final String value) {
-        try {
-            final String result  = String.format(format, value);
-            if (!result.equals(format) && !TextUtils.isEmpty(value)) {
-                return result;
-            }
-        } catch (IllegalFormatException ignored) { }
-        return format + (format.length() > 0 ? " " : "") + value;
+        return (format + (format.contains("%s") ? "" : " %s")).replace("%s", value);
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/activity/DocumentShareIntoFragment.java:119]: Edge cases either way. Went with simpler solution

**Comment by gsantner** [comment]
seems like this broke "shareinto - todo.txt". 

as it generally applies the prefix/format on the input field, instead of "at append time".  
and the todo.txt field is only shown when the input is singleline (which makes sense).  
thus, need to fix it so format is applied again AT time of appending, not at the time of creating the inputfield

**Pull request merged**